File: php/testcode/pathtraver_034.php
<?php
require_once __DIR__ . '/shared.php';

define('ASSETS_DIR', '/var/app/assets/');

// vuln-code-snippet start php_pt_config_dir_constant
function pathtraver034(BenchmarkRequest $req): BenchmarkResponse {
    $filename = basename($req->param('file'));
    $content = file_get_contents(ASSETS_DIR . $filename); // vuln-code-snippet safe-line php_pt_config_dir_constant
    return BenchmarkResponse::ok($content);
}
// vuln-code-snippet end php_pt_config_dir_constant

File: ruby/testcode/pathtraver_034.rb
require_relative 'shared'
require 'pathname'

# vuln-code-snippet start ruby_pt_cleanpath_prefix_check
def read_cleanpath_check(req)
  base = '/app/files'
  p = Pathname.new(base).join(req.param('rel')).cleanpath
  raise 'traversal' unless p.to_s.start_with?(base)
  p.read # vuln-code-snippet safe-line ruby_pt_cleanpath_prefix_check
  BenchmarkResponse.ok(p.read)
end
# vuln-code-snippet end ruby_pt_cleanpath_prefix_check
